src/pages/blocks/Blocks.tsx

```diff
@@ -207,14 +206,48 @@ const Blocks: React.FC<MatchParams> = props => {
             return ''
           }}
         />
-        <div className="load-more-button-container">
-          <Button
-            disabled={blockState.isLoading}
-            primary={false}
-            onClick={(): void => loadMore()}
-          >
-            load more
-          </Button>
+        <div className="block-list-pagination">
+          <div>
+            <Pagination
+              aria-label="Pagination"
+              model={model}
+              backgroundColor={'#7d9fb1'}
+              borderRadius={'80px'}
+              marginTop={10}
+              padding={10}
+            >
+              <Pagination.Controls>
+                <Pagination.JumpToFirstButton aria-label="First" />
+                <Pagination.StepToPreviousButton aria-label="Previous" />
+                <Pagination.PageList>
+                  {({ state }) =>
+                    state.range.map(pageNumber => (
+                      <Pagination.PageListItem key={pageNumber}>
+                        <Pagination.PageButton
+                          aria-label={`Page ${pageNumber}`}
+                          pageNumber={pageNumber}
+                        />
+                      </Pagination.PageListItem>
+                    ))
+                  }
+                </Pagination.PageList>
+                <Pagination.StepToNextButton aria-label="Next" />
+                <Pagination.JumpToLastButton aria-label="Last" />
+              </Pagination.Controls>
+              <Pagination.AdditionalDetails color={'#cae0eb'}>
+                {({ state }) =>
+                  `${getVisibleResultsMin(
+                    state.currentPage,
+                    perPage,
+                  )}-${getVisibleResultsMax(
+                    state.currentPage,
+                    perPage,
+                    blockState.totalCount,
+                  )} of ${blockState.totalCount} results`
+                }
+              </Pagination.AdditionalDetails>
+            </Pagination>
+          </div>
```

Comment:
what do you think about creating a separate component for this and reuse it on other pages ?